cms-contentful/src/main/java/io/commercetools/sunrise/cms/contentful/ContentfulCmsService.java

```diff
@@ -1,17 +1,95 @@
+import com.contentful.java.cda.CDAEntry;
+import com.contentful.java.cda.SynchronizedSpace;
 import io.commercetools.sunrise.cms.CmsIdentifier;
 import io.commercetools.sunrise.cms.CmsService;
 
 import java.util.List;
 import java.util.Locale;
+import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
+import java.util.function.Predicate;
 
 public class ContentfulCmsService implements CmsService {
 
+    private Map<String, CDAEntry> entries;
+
+    ContentfulCmsService(Map<String, CDAEntry> entries) {
+        this.entries = entries;
+    }
+
     @Override
     public CompletionStage<Optional<String>> get(final List<Locale> locales, final CmsIdentifier cmsIdentifier) {
-        return null; // TODO
+        return CompletableFuture.supplyAsync(() -> getEntry(cmsIdentifier)
+                .map(entry -> getLocalizedField(locales, entry, cmsIdentifier.getFieldName()))
+                .orElseGet(Optional::empty));
+    }
+
+    // TODO create models
+    private Optional<CDAEntry> getEntry(CmsIdentifier cmsIdentifier) {
+        return entries.values()
+                .stream()
+                .filter(entryPredicate(cmsIdentifier))
+                .findFirst();
+    }
+
+    private Predicate<CDAEntry> entryPredicate(CmsIdentifier cmsIdentifier) {
+        return entry -> {
+            final String contentfulEntryType = entry.contentType().id();
+            final String contentfulEntryKey = entry.getField(entry.contentType().displayField());
+            return contentfulEntryType.equals(cmsIdentifier.getEntryType())
+                    && contentfulEntryKey.equals(cmsIdentifier.getEntryKey());
+        };
+    }
+
+    private Optional<String> getLocalizedField(List<Locale> locales, CDAEntry cdaEntry, String fieldName) {
+        Optional<Locale> localeOptional = getFirstSupportedLocale(locales, cdaEntry, fieldName);
+        Object cdaEntryField = localeOptional.map(locale -> {
+            cdaEntry.setLocale(locale.toLanguageTag());
+            return cdaEntry.getField(fieldName);
+        }).orElse(null);
+        return getContentAccordingToFieldType(cdaEntryField);
+    }
+
+    private Optional<Locale> getFirstSupportedLocale(List<Locale> locales,
+                                                     CDAEntry cdaEntry, String fieldName) {
+        final Map<String, Object> stringObjectMap = cdaEntry.rawFields();
+        Map<String, Object> contentMap = (Map<String, Object>) stringObjectMap.get(fieldName);
+        if (contentMap == null) {
+            return Optional.empty();
+        }
+        Set<String> localesFromEntry = contentMap.keySet();
+        return locales.stream()
+                .filter(locale -> localesFromEntry.contains(locale.toLanguageTag()))
+                .findFirst();
+    }
+
+    private Optional<String> getContentAccordingToFieldType(Object cdaEntryField) {
```

Comment:
As a good practice, use `final` keyword for all variables so that you ensure you are working with immutable objects. And use `@Nullable` to let the IDE help you with possible wrong null values, and help developers know the kind of values they can provide. You can optionally use `@NonNull` too to clarify it even further.